src/Lean/Meta/Tactic/Apply.lean

```diff
@@ -61,19 +61,38 @@ private def dependsOnOthers (mvar : Expr) (otherMVars : Array Expr) : MetaM Bool
       let otherMVarType ← inferType otherMVar
       return (otherMVarType.findMVar? fun mvarId => mvarId == mvar.mvarId!).isSome
 
-private def reorderNonDependentFirst (newMVars : Array Expr) : MetaM (List MVarId) := do
-  let (nonDeps, deps) ← newMVars.foldlM (init := (#[], #[])) fun (nonDeps, deps) mvar => do
+/-- Partitions the given mvars in to two arrays (non-deps, deps)
+according to whether the given mvar depends on other mvars in the array.-/
+private def partitionDependentMVars (mvars : Array Expr) : MetaM $ Prod (Array MVarId) (Array MVarId) :=
+  mvars.foldlM (init := (#[], #[])) fun (nonDeps, deps) mvar => do
     let currMVarId := mvar.mvarId!
-    if (← dependsOnOthers mvar newMVars) then
+    if (← dependsOnOthers mvar mvars) then
       pure (nonDeps, deps.push currMVarId)
```

Comment:
We are trying to consistently use `return` instead of `pure` whenever applicable.